src/cider/piggieback_impl.clj

```diff
@@ -203,12 +203,41 @@
                    (readers/source-logging-push-back-reader
                     (java.io.StringReader. form-str))))))
 
+(defn- wrap-pprint [form]
+  `(let [sb# (goog.string.StringBuffer.)
+         sbw# (cljs.core/StringBufferWriter. sb#)
+         form# ~form]
+     (cljs.pprint/pprint form# sbw#)
```

Comment:
The problem with this approach is that we're effectively hardcoding pretty-printing, and nREPL goes to great lengths to ensure that printing is flexible and you can control what exactly do you want to print and how. I think it would be better if we made piggieback respect the nREPL infrastructure instead of forcing one particular way of pretty printing it in.

Reply:
What was the downside of reading on the clojure side and allowing nrepl to print it?